dynamiq/memory/backend/pinecone.py

```diff
@@ -0,0 +1,126 @@
+import uuid
+
+from pinecone import Pinecone as PineconeClient
+from pinecone import ServerlessSpec
+
+from dynamiq.components.embedders.base import BaseEmbedder
+from dynamiq.connections import Pinecone as PineconeConnection
+from dynamiq.memory.backend.base import Backend
```

Comment:
should we name it MemoryBackend? so it's not confused with anything else in the future?

Reply:
This is similar to our caching module. Initially, we used the term 'storage,' but then we switched to 'backend.'

Reply:
@acoola I suggest MemoryBackend

Reply:
@MaksymAI @delamainer @acoola @maksDev123 Let's vote on the final name, please.